plugin/src/main/kotlin/sc/plugin2024/Segment.kt

```diff
@@ -183,7 +183,16 @@ internal fun generateBoard(): Segments {
     val passengerTiles = shuffledIndices(PluginConstants.NUMBER_OF_SEGMENTS - 2, PluginConstants.NUMBER_OF_PASSENGERS).toArray()
     (2..PluginConstants.NUMBER_OF_SEGMENTS).forEach { index ->
         val previous = segments.last()
-        val direction = if(index == 2) CubeDirection.RIGHT else previous.direction.withNeighbors().filter { it != segments.takeLast(3).first().direction.opposite() }.random() // Do not allow three consecutive turns in one direction to prevent clashes
+        val direction = if(index == 2) CubeDirection.RIGHT else previous.direction
+                .withNeighbors()
+                // Do not allow four consecutive turns in one direction to prevent clashes
+                .filter {
+                    val lastSegments = segments.takeLast(4)
+                    
+                    previous.direction != lastSegments.first().direction.opposite()
+                            && it != lastSegments[1].direction.opposite()
```

Comment:
Doesn't this have to be an OR to allow the U-shape? And are you sure a U-shape cannot create problems? I have no problem preventing a U-shape. And yes as you said, this should be tested - because with this variation I am also less confident about it.

Reply:
My thoughts about this were that if both of the directions were the opposite, then a circular shape is guaranteed, otherwise it would be U-shaped. I tested a lot of scenarios with some sketches and I couldn't think of a combination of 2 right-headed segments + 6 U-shaped segments that created a clash with this approach. I would be very grateful if you wrote the unit tests since I don't know Kotlin very well.

Reply:
But you are right, it should be an OR statement.

Reply:
![screenshot-000090](https://github.com/software-challenge/backend/assets/30385979/4661633e-c5ea-462c-b0d4-1d0b62846351)

We don't want the red segments to be possible and the green segments should be possible. The orange segments are not possible with your approach although they would totally work out if we have a maximum of 6 turning segments.